gap/isomorph.gi

```diff
@@ -561,6 +612,140 @@ function(n, partition, method)
                 "form, <partition[i]> is the list of vertices with colour i,");
 end);
 
+InstallGlobalFunction(DIGRAPHS_ValidateEdgeColouring,
+function(graph, edge_colouring)
+  local n, colours, m, map, adji, coli, range, colour, mult, new_edge_colouring,
+  new_adj_list, count, seen_first, adj_colours, w, orientation_double, i, x,
+  edge, j;
+  #check: shapes and values from [1 .. something]
+  
+  if not IsDigraph(graph) then
+    ErrorNoReturn("the 1st argument must be a digraph"); 
+  fi;
+  n := DigraphNrVertices(graph);
+  if not IsList(edge_colouring) or Length(edge_colouring) <> n then
+    ErrorNoReturn("the 2nd argument must be a list of the same shape as",
+                  "OutNeighbours(graph), where graph is the 1st argument");
+  fi;
+  if ForAny(DigraphVertices(graph), x -> not IsList(edge_colouring[x]) or 
+                                         (Length(edge_colouring[x]) <>
+                                          Length(OutNeighbours(graph)[x]))) then
+    ErrorNoReturn("the 2nd argument must be a list of the same shape as",
+                  "OutNeighbours(graph), where graph is the 1st argument");
+  fi;
+
+  colours := [];
+  for adj_colours in edge_colouring do
+    for i in adj_colours do
+      if not IsPosInt(i) then
+        ErrorNoReturn("the 2nd argument should be a list of lists of",
+                      "positive integers");
+      fi;
+      AddSet(colours, i);
+    od;
+  od;
+  m := Length(colours);
+  if ForAny([1 .. m], i -> i <> colours[i]) then
```

Comment:
It won't make much difference, but you could write `if colours = [1..m] ...` here.